src/library/trackset/crate/cratefeature.cpp

```diff
@@ -862,7 +862,10 @@ void CrateFeature::storePrevSiblingCrateId(CrateId crateId) {
             TreeItem* pTreeItem = m_pSidebarModel->getItem(newIndex);
             DEBUG_ASSERT(pTreeItem != nullptr);
             if (!pTreeItem->hasChildren()) {
-                m_prevSiblingCrate = crateIdFromIndex(newIndex);
+                CrateId prevCrateId = crateIdFromIndex(newIndex);
+                if (prevCrateId.isValid()) {
```

Comment:
I did a look around and I have not yet understand in which cases it can be invalid at all? Does this case source from another bug deeper in the code? Can we have children with invalid crates? Please add a comment to the code.

Idea: The loop can be reversed and exited early with break. That would make the "search for one sibling" nature more clear.

Reply:
True, this is excessive. The check in `activateCrate()` is sufficient.
I'll revert this.